M2/Macaulay2/m2/newring.m2

```diff
@@ -77,7 +92,11 @@ tensor(QuotientRing,   QuotientRing) := monoidTensorDefaults >> optns -> (R, S)
      fg := substitute(f,(vars AB)_{0 .. m-1}) | substitute(g,(vars AB)_{m .. m+n-1});
      -- forceGB fg;  -- if the monomial order chosen doesn't restrict, then this
                      -- is an error!! MES
-     AB/image fg)
+     RS := AB/image fg;
+     promoteFromMap map(RS,R,(vars AB)_{0 .. m-1});
+     if S =!= R then promoteFromMap map(RS,S,(vars AB)_{m .. m+n-1});
```

Comment:
Is it natural that when `R == S` we favor the first ring map over the second?

Reply:
yeah this is annoying. no there isn't. in fact there are 3 choices here:
- favour the first
- favour the second
- decide that it's ambiguous and not call `promoteFromMap` at all.

I'm not sure what's best.

Reply:
I wonder if there's a general lesson to be learned from the way NormalToricVarieties handles natural maps:
```m2
i1 : needsPackage "NormalToricVarieties";

i2 : X = toricProjectiveSpace 1;

i3 : Y = toricProjectiveSpace 2;

i4 : Z = X ** Y;

i5 : Z_[0]

o5 = | 1 |
     | 0 |
     | 0 |

o5 : ToricMap Z <--- X

i6 : Z_[1]

o6 = | 0 0 |
     | 1 0 |
     | 0 1 |

o6 : ToricMap Z <--- Y

i7 : Z^[1]

o7 = | 0 1 0 |
     | 0 0 1 |

o7 : ToricMap Y <--- Z

i8 : Z^[0]

o8 = | 1 0 0 |

o8 : ToricMap X <--- Z
```
I wonder if we could have a notation for specific ring maps.